include/ginkgo/core/matrix/permutation.hpp

```diff
@@ -110,28 +185,54 @@ class Permutation : public EnableLinOp<Permutation<IndexType>>,
      * @return the number of elements explicitly stored in the permutation
      * array.
      */
-    size_type get_permutation_size() const noexcept
-    {
-        return permutation_.get_num_elems();
-    }
+    [[deprecated("use get_size()[0] instead")]] size_type get_permutation_size()
+        const noexcept;
+
+    [[deprecated("permute mask is no longer supported")]] mask_type
+    get_permute_mask() const;
+
+    [[deprecated("permute mask is no longer supported")]] void set_permute_mask(
+        mask_type permute_mask);
 
     /**
-     * Get the permute masks
+     * Returns the inverse permutation.
      *
-     * @return  permute_mask the permute masks
+     * @return a newly created Permutation object storing the inverse
+     *         permutation of this Permutation.
      */
-    mask_type get_permute_mask() const { return enabled_permute_; }
+    std::unique_ptr<Permutation> invert() const;
 
     /**
-     * Set the permute masks
+     * Combines this permutation with another permutation via composition.
+     * The resulting permutation fulfills `result[i] = other[this[i]]`
+     * or `result = other * this` from the matrix perspective, which is
+     * equivalent to first permuting by `this` and then by `other`.
```

Comment:
Just to be more clear, I would also add:
```suggestion
     * equivalent to first permuting by `this` and then by `other`.
     * Combining permutations $P_1$ and $P_2$ with `P_1.combine(P_2)` 
     * performs the operation permute(permute(A, P_1), P_2).
```

Reply:
great suggestion!